Composer/packages/client/src/components/AppUpdater/index.tsx

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React, { useContext, useEffect, useMemo, useState, useCallback } from 'react';
+import { Dialog, DialogFooter, DialogType } from 'office-ui-fabric-react/lib/Dialog';
+import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
+import { ProgressIndicator } from 'office-ui-fabric-react/lib/ProgressIndicator';
+import { ChoiceGroup } from 'office-ui-fabric-react/lib/ChoiceGroup';
+import formatMessage from 'format-message';
+
+import { StoreContext } from '../../store';
+import { AppUpdaterStatus } from '../../constants';
+
+import {
+  dialogContent,
+  dialogCopy,
+  dialogFooter,
+  modal,
+  optionRoot,
+  optionIcon,
+  updateAvailableDismissBtn,
+} from './styles';
+
+const { ipcRenderer } = window as any;
+
+function SelectOption(props) {
+  const { checked, text, key } = props;
+  return (
+    <div key={key} css={optionRoot}>
+      <Icon iconName={checked ? 'RadioBtnOn' : 'RadioBtnOff'} css={optionIcon(checked)} />
+      <span>{text}</span>
+    </div>
+  );
+}
+
+const downloadOptions = {
+  downloadOnly: 'downloadOnly',
+  installAndUpdate: 'installAndUpdate',
+};
+
+export const AppUpdater: React.FC<{}> = _props => {
+  const {
+    actions: { setAppUpdateError, setAppUpdateProgress, setAppUpdateShowing, setAppUpdateStatus },
+    state: { appUpdate },
+  } = useContext(StoreContext);
+  const { downloadSizeInBytes, error, progressPercent, showing, status, version } = appUpdate;
+  const [downloadOption, setDownloadOption] = useState(downloadOptions.installAndUpdate);
+
+  const handleDismiss = useCallback(() => {
+    setAppUpdateShowing(false);
+    if (status === AppUpdaterStatus.UPDATE_UNAVAILABLE || status === AppUpdaterStatus.UPDATE_FAILED) {
+      setAppUpdateStatus({ status: AppUpdaterStatus.IDLE, version: undefined });
+    }
+  }, [showing, status]);
+
+  const handlePreDownloadOkay = useCallback(() => {
+    // notify main to download the update
+    setAppUpdateStatus({ status: AppUpdaterStatus.UPDATE_IN_PROGRESS });
+    ipcRenderer.send('app-update', 'start-download');
+  }, []);
+
+  const handlePostDownloadOkay = useCallback(() => {
+    setAppUpdateShowing(false);
+    if (downloadOption === downloadOptions.installAndUpdate) {
+      ipcRenderer.send('app-update', 'install-update');
+    }
+  }, [downloadOption]);
+
+  const handleDownloadOptionChange = useCallback((_ev, option) => {
+    setDownloadOption(option);
+  }, []);
+
+  // listen for app updater events from main process
+  useEffect(() => {
+    ipcRenderer.on('app-update', (_event, name, payload) => {
```

Comment:
Does this work in the browser?

Reply:
Ah, good catch! I need to hide this component in the web app version. 😅